lib/date_range.rb

```diff
@@ -0,0 +1,14 @@
+class DateRange
+  attr_reader :check_in, :check_out
+  
+  def initialize(check_in:, check_out:)
+    @check_in = check_in
+    @check_out = check_out
+    
+    if check_out < check_in || check_in == check_out
+      raise Exception.new("Invalid date range provided; reservation cannot be made")
```

Comment:
If this class has no methods, does it need to be a class? Or does it _have_ behavior that you're not implementing here?